### DISCUSSION TITLE
Dark screen / no blink on RGB LED / on ESP32-3248S035C BOARD

### DISCUSSION THREAD

**Discussion by baeosk103** [rzeldent/esp32-smartdisplay#202] [General]
HI 

**my issue** : the ESP32-3248S035C BOARD can't run with the esp32-smartdisplay-demo code.
**help need** : how to make this board run as proper

I am very new on the  platformio and  esp32-smartdisplay/sunton. I try clone this link https://github.com/rzeldent/esp32-smartdisplay-demo and I keep them in my local Github folder "esp32-smartdisplay-demo". I  open VSC-platformio and open the project by choose the    "esp32-smartdisplay-demo" folder which I downloaded . Every thing look good for me but the vsc-platformio  show only  esp32-1932S019C file in the  folder "../.pio/libdeps/" after I open the project for few minute.

I think there is something wrong ,There is no ESP32-3248S035C board which I plan to test on the demo  code.

Therefore I think that I should select my boards ESP32-3248S035S in the the .ini file.

So  I  change some line in  the platformio.ini file at the project level by remove the # from that line to be "default envs = esp32-3248S035C". 

In the SRC/main.cpp I add  "#define BOARD_HAS_RGB_LED 1 and #define BAORD_HAS_CDS  1" on top of main.cpp. I build the project again at this moment there are the additional folder "esp32-3248S035C" made  in the .pio/build and .pio/libdeps .There are somes file inside these 2  folders.

After that I think  they should be OK to go  so I upload to my board, anyway there are nothing display("dark") on the screen , there are no RGB led blinks and they can't show any QR pic, flag pic or any CDS sensor value.

I need  help/advise from the community  how can I make this board work properly. 

Note : I test this board alone by use the TFT_ESPI basic graphic on arduino ,the display work well and all RGB also work well. 

//////////////////////////
**other test** : I test the cloned esp32-smartdisplay-demo  with the other  boards  by  remove # from  the platformio.ini file.  for  the esp32-2432S028Rv3 board  and the esp32-4880S07C board  ,both of  them can build , upload and run as the demo code should be. 

For the esp32-2432S028Rv3 , it show the RGB-LED blink and the CDS also work in order when I put the hand on the sensors, the sensor value was changed . The display show the count button , rotate button , QR code pic  and the flag pic , the screen can rotate when I touch the rotate button.

///////////////////////////////////////////////
**my platformio.ini file** 

; PlatformIO Project Configuration File
;
;   Build options: build flags, source filter
;   Upload options: custom upload port, speed and extra flags
;   Library options: dependencies, extra library storages
;   Advanced options: extra scripting
;
; Please visit documentation for the other options and examples
; https://docs.platformio.org/page/projectconf.html

[platformio]
#default_envs = esp32-1732S019C
#default_envs = esp32-1732S019N
#default_envs = esp32-2424S012C
#default_envs = esp32-2424S012N
#default_envs = esp32-2432S022N
#default_envs = esp32-2432S022C
#default_envs = esp32-2432S024C
#default_envs = esp32-2432S024N
#default_envs = esp32-2432S024R
#default_envs = esp32-2432S028R
#default_envs = esp32-2432S028Rv2
#default_envs = esp32-2432S028Rv3
#default_envs = esp32-2432S032C
#default_envs = esp32-2432S032N
#default_envs = esp32-2432S032R
default_envs = esp32-3248S035C
#default_envs = esp32-3248S035R
#default_envs = esp32-4827S043C
#default_envs = esp32-4827S043N
#default_envs = esp32-4827S043R
#default_envs = esp32-4848S040CIY1
#default_envs = esp32-4848S040CIY3
#default_envs = esp32-8048S043C
#default_envs = esp32-8048S043N
#default_envs = esp32-8048S043R
#default_envs = esp32-8048S050C
#default_envs = esp32-8048S050N
#default_envs = esp32-8048S050R
#default_envs = esp32-8048S070C
#default_envs = esp32-8048S070N
#default_envs = esp32-8048S070R
#default_envs = esp32-s3touchlcd7

[env]
platform = espressif32
framework = arduino

monitor_speed = 115200
monitor_rts = 0
monitor_dtr = 0
monitor_filters = esp32_exception_decoder

build_flags =
    -Ofast
    -Wall
    '-D BOARD_NAME="${this.board}"'
    '-D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_INFO'
    #'-D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_DEBUG'
    #'-D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_VERBOSE'
    '-D LV_CONF_PATH=${platformio.include_dir}/lv_conf.h'
    
lib_deps =
    https://github.com/rzeldent/esp32-smartdisplay#develop

[env:esp32-1732S019C]
board = esp32-1732S019C

[env:esp32-1732S019N]
board = esp32-1732S019N

[env:esp32-2424S012C]
board = esp32-2424S012C

[env:esp32-2424S012N]
board = esp32-2424S012N

[env:esp32-2432S022N]
board = esp32-2432S022N

[env:esp32-2432S022C]
board = esp32-2432S022C

[env:esp32-2432S024C]
board = esp32-2432S024C

[env:esp32-2432S024N]
board = esp32-2432S024N

[env:esp32-2432S024R]
board = esp32-2432S024R

[env:esp32-2432S028R]
board = esp32-2432S028R

[env:esp32-2432S028Rv2]
board = esp32-2432S028Rv2

[env:esp32-2432S028Rv3]
board = esp32-2432S028Rv3

[env:esp32-2432S032C]
board = esp32-2432S032C

[env:esp32-2432S032N]
board = esp32-2432S032N

[env:esp32-2432S032R]
board = esp32-2432S032R

[env:esp32-3248S035C]
board = esp32-3248S035C

[env:esp32-3248S035R]
board = esp32-3248S035R

[env:esp32-4827S043C]
board = esp32-4827S043C

[env:esp32-4827S043R]
board = esp32-4827S043R

[env:esp32-4827S043N]
board = esp32-4827S043N

[env:esp32-4848S040CIY1]
board = esp32-4848S040CIY1

[env:esp32-4848S040CIY3]
board = esp32-4848S040CIY3

[env:esp32-8048S043C]
board = esp32-8048S043C

[env:esp32-8048S043N]
board = esp32-8048S043N

[env:esp32-8048S043R]
board = esp32-8048S043R

[env:esp32-8048S050C]
board = esp32-8048S050C

[env:esp32-8048S050N]
board = esp32-8048S050N

[env:esp32-8048S050R]
board = esp32-8048S050R

[env:esp32-8048S070C]
board = esp32-8048S070C

[env:esp32-8048S070N]
board = esp32-8048S070N

[env:esp32-8048S070R]
board = esp32-8048S070R

[env:esp32-s3touchlcd7]
board = esp32-s3touchlcd7

[env:esp32-2432W328C]
board = esp32-2432W328C

[env:esp32-8048S550C]
board = esp32-8048S550C

////////////////////////////////////////////////////////////
**Some result after build**
Resolving esp32-3248S035C dependencies...
Library Manager: Installing git+https://github.com/rzeldent/esp32-smartdisplay#develop
git version 2.46.0.windows.1
Cloning into 'C:\Users\Admin\.platformio\.cache\tmp\pkg-installing-ciua22_y'...

Submodule 'boards' (https://github.com/rzeldent/platformio-espressif32-sunton.git) registered for path 'boards'

Cloning into 'C:/Users/Admin/.platformio/.cache/tmp/pkg-installing-ciua22_y/boards'...

Submodule path 'boards': checked out 'ae8d611a0ed8dacc744246bcaabd6711ccf793e9'
Library Manager: esp32_smartdisplay@3.0.0+sha.08d0ce2 has been installed!
Library Manager: Resolving dependencies...
Library Manager: Installing lvgl/lvgl @ ^9.2.0
Unpacking 0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Library Manager: lvgl@9.2.0 has been installed!
Updating metadata for the vscode IDE...
Project has been successfully updated!

//////////////////////////////////
 **the response from boards on serial**

--- Terminal on com10 | 115200 8-N-1
--- Available filter  ............. bababa 
---- More detail ...... bababa
---- Quit.......bababa
[5030][I][main.cpp:30] setup(): Boards;esp32-3248S035C
[5036][I][main.cpp:31] setup(): CPU;ESP32-D0WD-V3 rev3,CPU Freq 240 MHz,
 2  cores
[5044][I][main.cpp:32] setup(): Freeheap : 277652 bytes
[5050][I][main.cpp:33] setup(): FreePSRAM : 0 bytes
[5054][I][main.cpp:34] setup(): SDK version : v4.4.7-dirty

**Comment by van-persic** [comment]
I Have the same issue, but worse, because I don't get a serial response

**Comment by rzeldent** [comment]
Hi,

First of all, the ```platformio.ini``` is outdated (or develop branch). Take a look at https://github.com/rzeldent/esp32-smartdisplay/blob/main/platformio.ini. 
By default, all the boards are built. If you want only your board to be compiled, you can uncomment your board at the top of the ```platformio.ini``` file.

However, the esp32-smartdisplay is a library. This is only to test if it compiles. You have to include it somewhere to use it.
Please take a look at the https://github.com/rzeldent/esp32-smartdisplay-demo project to see how to use it...

Goog luck!

**Reply by baeosk103** [reply]
Hi rzeldent 

Great for your advise . 

So far I  copy the source code of main.cpp and  their ui from the esp32-smartdisplay-demo which I loaded from your link , I can see the board esp32-3248S035C in the .pio , I can build the project  and upload  to my board but there are no display show on LCD. 

I did  for the esp32 -2432S028Rv3 and esp32-4880S07C boards which I have  on hands, both of them worked well so I just wonder for the esp32-3248S035C  board.       

Yesterday I tried another way by  build the new project copying the esp32-3248S035C.json to the my local folder .platformio/platform/boards , I could build , upload but the result are the same , there are no display, no rgb blink. 

I will back  and test as your guide  again. 
 
Do anybody  have any experience work well on this board model? 

Thanks

**Comment by baeosk103** [comment]
Hi rzeldent

I try re-experiment  from the green field again  by
A. Starting test with esp32-2423S028Rv3  
- I download with github-desktop program from the  esp32-smartdispaly-demo links. 
- I move this download folder to the project folder inside the vsc-platformio. 
- I start copy the board definition for the esp32-2432S028Rv3.json into the boards folder in the vsc-platformio  
- I remove # from the platformio.ini  file to activate only the esp32-24-32S028Rv3.
- After the few  minutes the  the system start config the project  and  generate the pio folder which has the build folder and  libdeps folder.
- I add the #define BOARD_USE_RGB_LED 1 and #define BAORD_HAS_CDS 1 in the main.cpp in the src folder of  my project to show the features on my boards.  
- I build the project base on esp32-2432s028Rv3 board, It took  me several minutes but they are built successful.
- I upload  to my board esp32-2432s028Rv3 , the result is quite good as my attached photo.

B. Start testing with ESP32-3248S035C board
 I start from the green field  ,the same step which I did for the ESP32-2432S028RV3.
 I can't see any response from the experiment , I capture the result in the attached photo
![IMG_20241002_114826](https://github.com/user-attachments/assets/a294c56d-79bd-4920-bf05-c60063dbb9d3)
  
Hi rezeldent  can you advise which process/step or the config which I did wrong between I try experiment with esp32-3248S035C board.

Thanks a lot

#"my dream will come true"#

**Comment by rzeldent** [comment]
Take a look here: https://github.com/rzeldent/esp32-smartdisplay-demo/issues/43

**Reply by baeosk103** [reply]
rezeldent

your reply is great, I change the size of buffer  as your guide "'-D LVGL_BUFFER_PIXELS=(DISPLAY_WIDTH*DISPLAY_HEIGHT/16)'", ,my board  work on the fly after I upload them.

I  think  that we can use 320*480/8 (for esp32-3248S035C/R.json) as well  since  it equivalent to the  240*320/4(esp32-2432S028Rv3.json). They also work well in my test.    

Cheers.

#I belive ... you can make them fly#
![IMG_20241002_200411](https://github.com/user-attachments/assets/18afed6c-8a49-45dc-a88a-0237f9adafc4)

**Reply by TronCarter269** [reply]
> Take a look here: [rzeldent/esp32-smartdisplay-demo#43](https://github.com/rzeldent/esp32-smartdisplay-demo/issues/43)

This worked for me.  Now on to my next issues....
